### PR TITLE
Add resetState, reset to powerplug 

### PR DESCRIPTION
**Pull request by istarkov**
Ref #110

see [comment why](https://github.com/renatorib/react-powerplug/issues/110#issuecomment-411932761)

I've added in first commit eslint flow plugin as otherwise it's hard to edit `flow` file in editors like vscode (flow file is one big eslint error)

cc @TrySound  please check and i'll proceed with other Components

### CODE REVIEW

File: src/components/Value.js [hunk 1]
```diff
@@ -9,10 +9,11 @@ const Value = ({ initial, onChange, ...props }) => (
     initial={{ value: initial }}
     onChange={onChangeProp(onChange, 'value')}
   >
-    {({ state, setState }) =>
+    {({ state, setState, resetState }) =>
       renderProps(props, {
         value: state.value,
         set: value => setState(s => ({ value: set(value, s.value) })),
+        reset: () => resetState(),
```
Comment (TrySound) [inline src/components/Value.js]: Why not `reset: resetState`?

Comment (istarkov) [reply src/components/Value.js]: By analogy, State has setState with callback, so resetState has too, but all other components has no callback, so here we remove callback too

**Comment by istarkov** [comment]
The possible issues with current reset implementation is that if object Shape will be changed in initial,
setState inside reset will merge new Shape with previous instead reset.

To avoid we need to make instead of this

```
class State extends Component {
  state = {
    ...this.props.initial,
  }
```

this

```
class State extends Component {
  state = {
    value: { ...this.props.initial },
  }
```

It's fun but it was the first PR I made to react-redux ;-) moving state into value

**Comment by istarkov** [comment]
Fixed in next setState implementation

**Comment by istarkov** [comment]
BTW for my own case, I don't need that reset for every helper, and if we add Prop types to setState at State helper updater, I can reset State without `resetState` method, just using 

```
setState((_, {initial}) => initial)
```

So feel free to close the PR, as it so-so needed ;-) and for my case (having that I have stable object shapes) result can be achieved just making few type changes

**Comment by istarkov** [comment]
Close in favor of #151 

PS:
TODO:
Pick flow eslint from this PR and create new

**Pull request closed without merging**